pandas/__init__.py

```diff
@@ -202,8 +202,29 @@
         FutureWarning,
         stacklevel=2,
     )
-# Don't allow users to use pandas.os or pandas.warnings
-del os, warnings
+# Don't allow users to use pandas.os
+del os
+
+# DeprecationWarning for missing pyarrow
+from pandas.compat.pyarrow import pa_version_under10p1
+
+if pa_version_under10p1:
+    import warnings
+    from pandas.compat._optional import VERSIONS
+
+    warnings.warn(
+        f"""
+Pyarrow will become a future required dependency of pandas,
```

Comment:
It adds a bit of extra complexity, but I'd personally add another `if` and show different warnings when PyArrow is not installed and when it's too old. I think the displayed warning will be much more clear and concise. Just a personal opinion, no big deal if you prefer to keep it this way.

Reply:
Updated.